<center><h1> Statistical Test</h1></center>

___

<h2> t-test</h2>

t-distribution is a continuous probability distribution, derived from normal distribution, used when sample is too small to capture the population distribution. 

t-test is used in place of z-test when: 
* Population variance is not known 
* Sample size is small (< 30) 

<center><b>t = (x̅ − μ)/(s/√ n)</b></center>


<img src = "img/t-test.png">

### Question-1

A coffee shop relocates to Italy and wants to make sure that all lattes are consistent. They believe that each latte has an average of 4 oz of espresso. If this is not the case, they must increase or decrease the amount. A random sample of 25 lattes shows a mean of 4.6 oz of espresso and a standard deviation of .22 oz. Use alpha = .05 and run a one sample t-test and compare with the known population mean.


- Frame the hypothesis
    - Null Hypothesis: H0: mean= 40z
    - Alternate Hypothesis: H1: mean != 4oz
- Expected mean: 4 oz
- Observed: 4.6
- sd: 0.22
- samples: 25


In [1]:
mu = 4
n = 25
xbar = 4.6
s = 0.22
alpha = 0.05 # significance level

#lets calulate the test statistic

t_calc= (xbar-mu)/(s/sqrt(n))
t_calc

[1] 13.63636

In [2]:
## To calculate the critical regions
alpha=0.05

## since this is a two tailed test, the critical region lies on both sides.
left_critical<-qt(p = 0.025, df = n-1) ## critical region on left
right_critical<-qt(p = 0.025, df = n-1,lower.tail=F) ## critical region on right

left_critical
right_critical

[1] -2.063899

[1] 2.063899

Decision Making method 1: If the t_calculated lies between the two critical values then its ok
Else the t lies in critical region
Observe that 13.63> 2.06 i.e lies in critical region. If value lies in critical region then we reject the null hypothesis

In [20]:
## Method2:
p_cal= pt(t_calc,df=n-1,lower.tail = T)
significance_level=0.05/2
p_cal

[1] 0.9999996

In [12]:
100-0.025

[1] 99.975

 Here pvalue<significance_level hence reject the null hypothesis

### Question-2

An outbreak of Salmonella-related illness was attributed to ice cream produced at a certain factory. Scientists measured the level of Salmonella in 9 randomly sampled batches of ice cream. The levels (in MPN/g) were: 0.593, 0.142, 0.329, 0.691, 0.231, 0.793, 0.519, 0.392, 0.418. Is there evidence that the mean level of Salmonella in the ice cream is greater than 0.3 MPN/g?



- Frame the hypothesis
    - Null Hypothesis: H0: mean <= 0.3 MPN/g
    - Alternate Hypothesis: H1: mean > 0.3 MPN/g
- Expected mean: <=0.3 MPN/g
- Observed: To be calculated
- sd: To be calculated
- samples: 9

In [23]:
(mean(sal_level)-0.3)/(sd(sal_level)/sqrt(9))
qt(p = 0.05, df = 8,lower.tail=F)

[1] 2.205059

[1] 1.859548

In [4]:
sal_level = c(0.593, 0.142, 0.329, 0.691, 0.231, 0.793, 0.519, 0.392, 0.418)
mu=0.3
n = 9
## Performing t-test
t_calculated=t.test(sal_level, mu=0.3, conf.level = 0.95, alternative = "greater")
t_calculated

t_calculated$statistic



	One Sample t-test

data:  sal_level
t = 2.2051, df = 8, p-value = 0.02927
alternative hypothesis: true mean is greater than 0.3
95 percent confidence interval:
 0.3245133       Inf
sample estimates:
mean of x 
0.4564444 


t 
2.205059

In [5]:
significance_level=0.05
df=n-1
df

t_critical= qt(significance_level,df,lower.tail = FALSE)
t_critical

[1] 8

[1] 1.859548

Observation: Since t_calculated is greater than t_critical (2.2051 > 1.859548), the t-score lies in the critical region, hence we reject the Null hypothesis.

### Question-3

Do two types of music, type-I and type-II have different effects on student’s ability to perform a series of 40 tasks that requires concentration? Number of topics completed by each student under both environments is listed below. The two types of music are independent of each other and assume that the sampling distribution follows a normal distribution. Can you support the claim that type-I music improved the students’ performance over type-II music? 

- Frame the hypothesis
    - H0: (scores_with_type1) - (scores_with_type2) = 0
    - H1: (scores_with_type1) - (scores_with_type2) > 0


In [6]:
type1 = c(21, 22, 18, 19, 26, 22, 25, 25, 24, 21, 23, 23, 26, 29, 22)
type2 = c(21, 23, 21, 20, 21, 29, 20, 16, 20, 17, 21, 25, 26, 18, 19)

In [7]:
t_stat = t.test(type1, type2, mu=0, conf.level = 0.95, alternative='greater')
t_stat


	Welch Two Sample t-test

data:  type1 and type2
t = 1.6723, df = 27.112, p-value = 0.05298
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 -0.03550643         Inf
sample estimates:
mean of x mean of y 
 23.06667  21.13333 


In [8]:
df=as.integer(t_stat$parameter)
df
t_critical = qt(0.05, df=df, lower.tail = F)
t_critical

[1] 27

[1] 1.703288

Observation: Since t statistic value is less than t critical, we fail to reject the null hypothesis. Staitistically, The two types of music do not have different effects upon the ability of college students to perform a series of mental tasks requiring concentration.